torch/_dynamo/output_graph.py

```diff
@@ -1413,13 +1413,103 @@ def example_inputs(self) -> List[torch.Tensor]:
+            from torch.fx.experimental.sym_node import SymTypes
+
+            if isinstance(a, (int, float, bool)):
+                return True
+            if isinstance(a, fx.Node):
+                return isinstance(a.meta.get("example_value"), SymTypes)
+            return False
+
+        # NB: We assume that you cannot do mutations on int/float/bool,
+        # because they are immutable types, and therefore is always safe to
+        # DCE.
+        def is_symnode_compute_node(node):
+            from torch.fx.experimental.sym_node import SymTypes
+
+            if node.op != "call_function":
+                return False
+            # TODO: I don't think it's possible to have a bare int/float here?
+            if not isinstance(node.meta.get("example_value"), SymTypes):
+                return False
+            # TODO: This will bail here if you ever end up with a more complicated
+            # computation function, like sum(list_of_ints), even though it
+            # should be DCE'able
+            if not all(is_symnode_arg(a) for a in node.args):
+                return False
+            if not all(is_symnode_arg(a) for a in node.kwargs.values()):
+                return False
+            return True
+
+        # NB: You could try to expand this to cover more cases by simply
+        # detecting whenever you have an int output, but this is a bit
+        # dangerous in case someone adds a function that returns an int but is
+        # mutating.  So manually whitelist for now.
+        def is_accessor_node(node):
+            if (
+                node.op == "call_method"
+                and isinstance(node.args[0].meta.get("example_value"), torch.Tensor)
+                and node.target in ["size", "stride", "storage_offset", "item"]
+            ):
+                return True
+            if node.op == "call_function" and node.target in [
+                torch.ops.aten.sym_size,
+                torch.ops.aten.sym_size.default,
+                torch.ops.aten.sym_size.int,
+                torch.ops.aten.sym_stride,
+                torch.ops.aten.sym_stride.default,
+                torch.ops.aten.sym_stride.int,
+                torch.ops.aten.sym_storage_offset,
+                torch.ops.aten.sym_storage_offset.default,
+            ]:
+                return True
+            return False
+
         for node in reversed(list(self.graph.nodes)):
             if len(list(node.users)) == 0:
-                if node.op == "get_attr":
-                    self.remove_node(node)
-                elif node.op == "call_function" and node.target is operator.getitem:
+                if (
+                    node.op == "get_attr"
+                    or (node.op == "call_function" and node.target is operator.getitem)
+                    or (
+                        node.op == "call_function"
+                        and node.target is torch._check
+                        and is_static_true(node.args[0])
```

Comment:
Can we ever hit this? Wouldn't these be elided before via our evaluate_static function?

Reply:
For whatever reason, we don't elide tracing in torch._check in Dynamo. Though, arguably, we should!